### DISCUSSION TITLE
Can Convert.ToBoolean supports "1" and "0" values?

### DISCUSSION THREAD

**Discussion by mertmtn** [dotnet/runtime#86586] [General]
Existing version of Convert.ToBoolean, only convert from below values.
1,0, "True", "False", "true","false"

We have to apply extra conversion method Convert.ToInt32 in order to convert "1" and "0" to true and false respectively.

Also we have to write extra piece of code like below.
`bool boolValue = (convertValue== "1")`

Shall we need to improve the method without writing any extension method or using above solutions?
Perhaps, an improvement can be made for the problem as future works

**Comment by HaloFour** [comment]
Suggestions for API changes should go to the [runtime](https://github.com/dotnet/runtime) repository.

**Reply by mertmtn** [reply]
Opened a new issue as your feedback. https://github.com/dotnet/runtime/issues/86579

**Comment by DavidArno** [comment]
It would be a huge mistake in my view if the runtime mandated that `"1"` convert to `true`. Process return values for example return `0` for success, so I might want that mapped to `true`.

Such mappings are clearly domain specific and therefore the correct place for them to exist is in your domain logic, not within the BCL.

**Reply by FaustVX** [reply]
This is valid and non-valid at the same time.
In the .NET world (like most C-based languages), `false` map to `0` and `true` to other values.
Look at this code [sharplab.io](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4AEBMBGAWACgsAGAAizwDoAldBKMAU2oEkYFnkB7ABwDK3AG5QAxswDOAbmLEqATgAUMZgHdyAYR4xhSgJQBvcsHIBechQC+1CPtkk8y1Ru26Dx0xbzkbdh4oq6lo6ekYm5uQArL629nJEANoCCGhiCAAyEACePKgISlm5+QDSsAAm1ACiAB58ADbiUAj6ALrEkqmo6SG6xIbE5EPkiQBiUMz15QDyAGazkswFpG2Dw1gAzCY8PPXkEA7DI+OTM/OLy6tER5sm2VwmDlZAA==)

**Reply by mertmtn** [reply]
In my opinion, If 1 as integer can convert to true, "1" must be convertible.

**Reply by HaloFour** [reply]
I disagree.  Convertibility is not transitive.  I think in cases like this it wades into "truthiness" territory.  Like what about "00" or "01" or "2"?  I think it opens a can of worms to attempt to interpret a string as a number in order to interpret it as a Boolean.

**Comment by cansozbir** [comment]
I would expect Convert.ToBoolean to return 0 for an empty string, and 1 for any other string. 
This is the current behavior of Python as well.

**Reply by mertmtn** [reply]
Perhaps, you could consider with another issue because of "empty string" case.

**Comment by CyrusNajmabadi** [comment]
Moving to runtime.
